src/github/repository.ts

```diff
@@ -49,6 +51,10 @@ export const deleteRepositoryWebhookHandler = async (context: WebhookContext, ji
 		transformRepositoryId(context.payload.repository?.id, context.gitHubAppConfig?.gitHubBaseUrl)
 	);
 
+	// Update the total number of repos, we can assume atleast 1 repo(just deleted repo) exists for the default value
+	const totalNumberOfRepos = ((subscription.totalNumberOfRepos || 1) - 1);
+	await subscription.update({ totalNumberOfRepos });
+	await RepoSyncState.deleteRepoForSubscription(subscription, context.payload.repository?.id);
 	webhookProcessComplete(context, jiraResponse?.status);
```

Comment:
Can we first delete from `RepoSyncState` before updating the `totalNumberOfRepos` in `subscription` table? I think, we also need to check whether any recored is deleted from `RepoSyncState` before updating count in `Subscription` table.